src/plugins/blackbox/public/js/blackbox.js

```diff
@@ -22,26 +22,36 @@
   head.appendChild(js);
   var Blackbox = function Blackbox(cockpit) {
     console.log('Loading Blackbox plugin.');
+    var self = this;
+
     this.cockpit = cockpit;
     this.recording = false;
     this.idb;
     this.sessionID = this.newSession();
     this.eventBuffer = [];
     this.otherBuffer = [];
     this.sessions_cache = [];
-  };
-  plugins.Blackbox = Blackbox;
-  Blackbox.prototype.inputDefaults = function inputDefaults() {
-    var self = this;
-    return [{
-        name: 'blackbox.record',
-        description: 'Start recording telemetry data.',
-        defaults: { keyboard: 'r' },
-        down: function () {
+
+    this.inputDefaults = [
+    {
+      name: 'blackbox.record',
+      shortName: "Toggle Blackbox recording",
+      description: 'Start recording the telemetry data.',
+      controllers: new Map([["keyboard", "r"]]),
```

Comment:
I wonder if there is a way to combine the controllers/actions.  It feels disconnected to call out the keyboard+key binding and then the activated function with an action.
What about:
```
bindings: [{
      controller:"keyboard",
      key:"r", 
      action:"down",
      activates:"toggleRecording"
}],
activations: [{toggleRecording: function(){...}]
```